cardano-node/src/Cardano/Node/Configuration/POM.hs

```diff
@@ -187,6 +194,12 @@ instance FromJSON PartialNodeConfiguration where
       pncTargetNumberOfEstablishedPeers <- Last <$> v .:? "TargetNumberOfEstablishedPeers"
       pncTargetNumberOfActivePeers      <- Last <$> v .:? "TargetNumberOfActivePeers"
 
+      -- Enable P2P switch
+      p2pSwitch <- v .:? "EnableP2P" .!= Just False
+      let pncEnableP2P =
+            case p2pSwitch of
+              Nothing -> mempty
+              Just r  -> Last $ Just (bool DisabledP2PMode EnabledP2PMode r)
```

Comment:
It will be cleaner to unroll the `bool`, and pattern match on `Just True` and `Just False`.